src/main/kotlin/solve/rendering/engine/core/renderers/FramesRenderer.kt

```diff
@@ -167,7 +168,9 @@ class FramesRenderer(
 
     private fun uploadLoadedFramesToBuffers() {
         bufferFramesToUpload.toList().forEach { frame ->
-            bufferFramesToUpload.remove(frame)
+            synchronized(bufferFramesToUpload) {
```

Comment:
you already have sync list

Reply:
> you already have sync list

I've decided to change the `synchronizedList` to `CopyOnWriteArrayList` because there is an iteration of elements